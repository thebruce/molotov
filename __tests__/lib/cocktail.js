const cocktailHelper = require('../helpers/cocktailHelper');
const Cocktail = require('../../lib/cocktail');
const { MolotovError } = require('../../lib/molotovError');
const supers = require('../helpers/supers');
const mixins = require('../helpers/mixins');
const cocktailConfig = {
  testMolotovImplementer: {
    supersNameSpace: {
      testSuper: 'testSuper'
    },
    molotovPlugins: {
      testSuper: {
        onlyPluginOne: [
          'pluginOne',
        ],
        pluginOneTwo: [
          'pluginOne',
          'pluginTwo',
          'pluginFour',
        ],
        pluginAll: [
          'pluginFour',
        ],
      },
    },
  },
};

let tmpMocks = [];

describe('Cocktail Basic tests', () => {
  beforeEach(() => {
    tmpMocks.forEach(mock => mock.mockRestore());
    tmpMocks = [];
    jest.resetAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  test('Cocktail constructor in action', () => {
    expect.assertions(3);
    expect(cocktailHelper.getCocktailSupers()).toMatchSnapshot();
    expect(cocktailHelper.getCocktailMixins()).toMatchSnapshot();
    expect(cocktailHelper.getCocktailConfig()).toMatchSnapshot();
  });
});

describe('Cocktail Constructor Errors', () => {
  beforeEach(() => {
    tmpMocks.forEach(mock => mock.mockRestore());
    tmpMocks = [];
    jest.resetAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  test('Bad config', () => {
    expect.assertions(3);
    expect(() => new Cocktail(null)).toThrowError(MolotovError);
    expect(() => new Cocktail({})).toThrowError(MolotovError);
    expect(() => new Cocktail('')).toThrowError(MolotovError);
  });
});

describe('Cocktail Constructor empty super', () => {
  test('Cocktail empty super', () => {
    expect.assertions(1);
    const cocktail = new Cocktail(cocktailConfig, {}, mixins);
    expect(cocktail.getCocktailSupers()).toBeUndefined();
  });
});

describe('Cocktail Constructor empty mixins', () => {
  test('Cocktail empty mixins', () => {
    expect.assertions(1);
    const cocktail = new Cocktail(cocktailConfig, supers, {});
    expect(cocktail.getCocktailMixins()).toBeUndefined();
  });
});