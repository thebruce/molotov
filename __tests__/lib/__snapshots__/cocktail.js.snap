// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cocktail Basic tests Cocktail constructor in action 1`] = `
Object {
  "testSuper": [Function],
  "testSuperTwo": [Function],
}
`;

exports[`Cocktail Basic tests Cocktail constructor in action 2`] = `
Object {
  "testSuper": Object {
    "pluginFour": [Function],
    "pluginThree": [Function],
  },
  "testSuperTwo": Object {
    "pluginSeven": [Function],
  },
}
`;

exports[`Cocktail Basic tests Cocktail constructor in action 3`] = `
Object {
  "testMolotovImplementer": Object {
    "molotovPlugins": Object {
      "testSuper": Object {
        "onlyPluginOne": Array [
          "pluginOne",
        ],
        "pluginAll": Array [
          "pluginFour",
        ],
        "pluginOneTwo": Array [
          "pluginOne",
          "pluginTwo",
          "pluginFour",
        ],
      },
      "testSuperTwo": Object {
        "pluginSeven": Array [
          "pluginSeven",
        ],
      },
    },
    "supersNameSpace": Object {
      "testSuper": "testSuper",
      "testSuperTwo": "testSuperTwo",
    },
  },
}
`;
